agents/web_agent.py

```diff
@@ -0,0 +1,508 @@
+    # Email input selectors
+    EMAIL_INPUTS = [
+        "input[type='email']",
+        "#identifierId",
+        "input[name='email']",
+        "input[autocomplete='username']"
+    ]
+    
+    # Email next/submit button selectors
+    EMAIL_NEXT_BUTTONS = [
+        "#identifierNext",
+        "button:has-text('Next')",
+        "input[type='submit']",
+        "button[type='submit']"
+    ]
+    
+    # Password input selectors
+    PASSWORD_INPUTS = [
+        "input[type='password']",
+        "input[name='password']",
+        "#password",
+        "input[autocomplete='current-password']"
+    ]
+    
+    # Password next/submit button selectors
+    PASSWORD_NEXT_BUTTONS = [
+        "#passwordNext",
+        "button:has-text('Next')",
+        "input[type='submit']",
+        "button[type='submit']"
+    ]
+
+
+class WebAgent(CodingAgent):
+    """Base class for web-based coding agents using Botright"""
+    
+    def __init__(self, computer_use_client):
+        super().__init__(computer_use_client)
+        self.botright_client = None
+        self.browser = None
+        self.page = None
+        self._is_browser_ready = False
+    
+    @property
+    def window_name(self) -> str:
+        """Web agents don't have traditional windows, but we keep this for compatibility"""
+        return f"{self.agent_name}_browser"
+    
+    @property
+    @abstractmethod
+    def web_url(self) -> str:
+        """The URL of the web-based coding agent"""
+        pass
+    
+    @property
+    @abstractmethod
+    def input_selector(self) -> str:
+        """CSS selector for the input field where prompts are entered"""
+        pass
+    
+    @property
+    @abstractmethod
+    def submit_selector(self) -> str:
+        """CSS selector for the submit button"""
+        pass
+    
+    @property
+    @abstractmethod
+    def output_selector(self) -> str:
+        """CSS selector for the output/response area"""
+        pass
+    
+    @property
+    @abstractmethod
+    def loading_selector(self) -> Optional[str]:
+        """CSS selector for loading indicator (optional)"""
+        pass
+    
+    @property
+    def interface_state_prompt(self) -> str:
+        """Not used for web agents, but kept for compatibility"""
+        return ""
+    
+    @property
+    def resume_button_prompt(self) -> str:
+        """Not used for web agents, but kept for compatibility"""
+        return ""
+    
+    @property
+    def input_field_prompt(self) -> str:
+        """Not used for web agents, but kept for compatibility"""
+        return ""
+    
+    def set_current_project(self, project_path: str):
+        """Override to handle web agent project context"""
+        super().set_current_project(project_path)
+        # Web agents might need to navigate to project-specific URLs or set context
+        # This can be customized by subclasses
+    
+    def set_repository_context(self, repo_url: str, original_repo_url: Optional[str] = None):
```

Comment:
fyi: this code might be irrelevant once we move into the new Git communication protocol (from my whatsapp recording)